examples/jsm/utils/BufferGeometryUtils.js

```diff
@@ -199,7 +202,22 @@ function mergeBufferGeometries( geometries, useGroups = false ) {
 		mergedGeometry.userData.mergedUserData = mergedGeometry.userData.mergedUserData || [];
 		mergedGeometry.userData.mergedUserData.push( geometry.userData );
 
-		if ( useGroups ) {
+		if ( useGroups && geometry.groups.length > 0 ) {
+
+			// Keep existing groups
+
+			for ( let group of geometry.groups ) {
+
+				let offsetMaterialIndex = groupIndex + group.materialIndex;
+				mergedGeometry.addGroup( offset, group.count, offsetMaterialIndex );
+				offset += group.count;
+				nextGroupIndex = Math.max( nextGroupIndex, offsetMaterialIndex );
+
+			}
```

Comment:
I'm not sure if this is correct. It looks like this code assumes that all groups are sequential and situated next to each other but that is not guaranteed. I believe group.start needs to be accounted for.

Reply:
Yep, good catch. I've adjusted this now to account for `group.start`